server-spi-private/src/main/java/org/keycloak/models/utils/ModelToRepresentation.java

```diff
@@ -230,16 +230,28 @@ public static GroupRepresentation toGroupHierarchy(GroupModel group, boolean ful
         return rep;
     }
 
-    private static boolean groupMatchesSearchOrIsPathElement(GroupModel group, String search, Boolean exact) {
+    public static boolean groupMatchesSearchOrIsPathElement(GroupModel group, String search, Boolean exact) {
```

Comment:
Could you please use `boolean` instead? It should make it simpler.

Reply:
just keept to original signature, and used the optional "exact" parameter passed down from `org.keycloak.models.utils.ModelToRepresentation#searchForGroupByName(org.keycloak.models.KeycloakSession, org.keycloak.models.RealmModel, boolean, java.lang.String, java.lang.Boolean, java.lang.Integer, java.lang.Integer)`
I'd keep that parameter Boolean and use boolean downstream.

Reply:
I see. If you agree we can make this are a bit better we should just use the primitive type instead.

Reply:
Yep, I just saw that the rest enpoint `org.keycloak.services.resources.admin.GroupsResource#getGroups` uses `Boolean exact`  with a default value, so it seems to be safe to use primitive boolean downstream.

Just updated the PR.

Reply:
Works for me. Thanks.